saleor/dashboard/menu/utils.py

```diff
@@ -92,3 +94,16 @@ def get_menus_that_needs_update(collection=None, categories=None, page=None):
     menus_to_be_updated = MenuItem.objects.filter(q).distinct().values_list(
         'menu', flat=True)
     return menus_to_be_updated
+
+
+def get_menu_obj_text(obj):
+    if obj.is_published:
+        return str(obj)
+    elif isinstance(obj, Page) and obj.is_visible and obj.available_on:
+        return pgettext(
+            'Menu item page hidden status',
+            'Hidden (will become visible on %s)' % (localize(
```

Comment:
For translations, we should always use a meaningful formatting. 

Translator won't know what goes under `%s`, formatting it as `%(available_on_date)s` will give a nice feel, of how it will look like on the real site.